toolchain/check/handle_class.cpp

```diff
@@ -345,23 +368,15 @@ auto HandleBaseDecl(Context& context, Parse::BaseDeclId node_id) -> bool {
 
   auto enclosing_class_decl = context.GetCurrentScopeAs<SemIR::ClassDecl>();
   if (!enclosing_class_decl) {
-    CARBON_DIAGNOSTIC(BaseOutsideClass, Error,
-                      "`base` declaration can only be used in a class.");
-    context.emitter().Emit(node_id, BaseOutsideClass);
+    DiagnoseClassSpecificDeclOutsideClass(context, node_id,
+                                          Lex::TokenKind::Base);
     return true;
   }
```

Comment:
Note the wrapping calls are also similar.

Reply:
Factored that out too.